packages/signer/src/ExternalSigner.js

```diff
@@ -0,0 +1,66 @@
+/* eslint-disable no-unused-vars */
+import * as pkijs from 'pkijs';
+import {SignPdfError} from '@signpdf/utils';
+import {Signer} from './Signer';
+
+/**
+ * Abstract ExternalSigner class taking care of creating a suitable signature for a given pdf
+ * using an external signature provider.
+ * Subclasses should specify the required signature and hashing algorithms used by the external
+ * provider (either through the `signAlgorithm` and `hashAlgorithm` attributes, or by overriding
+ * the `getSignAlgorithm` and `getHashAlgorithm` methods), as well as provide the used signing
+ * certificate and final signature (by implementing the `getCertificate` and `getSignature`
+ * methods).
+ */
+export class ExternalSigner extends Signer {
+    /**
+     * Method to retrieve the signature of the given hash (of the given data) from the external
+     * service. The original data is included in case the external signature provider computes
+     * the hash automatically before signing.
+     * To be implemented by subclasses.
+     * @param {Uint8Array} hash
+     * @param {Uint8Array} data
+     * @returns {Promise<Uint8Array>}
+     */
+    async getSignature(hash, data) {
+        throw new SignPdfError(
+            `getSignature() is not implemented on ${this.constructor.name}`,
+            SignPdfError.TYPE_INPUT,
+        );
+    }
+
+    /**
+     * Get a "crypto" extension and override the function used by SignedData.sign to support
+     * external signing.
+     * @returns {pkijs.ICryptoEngine}
+     */
+    getCrypto() {
+        const crypto = super.getCrypto();
+        crypto.sign = async (_algo, _key, data) => {
+            // Calculate hash
+            const hash = await crypto.digest({name: this.hashAlgorithm}, data);
+            // And pass it to the external signature provider
+            const signature = await this.getSignature(Buffer.from(hash), Buffer.from(data));
+            return signature;
+        };
+        return crypto;
+    }
```

Comment:
This is the reason for having this `getCrypto` method. But it should be considered a "private" method that is not needed for most `Signer` implementations.